Implement Slash Commands

RisingFog opened the pull request:
This PR implements the majority of the commands as slash commands, including the main verify command. If the $verify command is sent as a DM, it will inform the user to use the slash command version of verify instead. All slash command replies are now ephemeral, which means that only the user who initiates the command will see the reply from the bot.

![image](https://user-images.githubusercontent.com/2121359/163894523-88947178-1775-44d8-846a-286f139c80ae.png)

![image](https://user-images.githubusercontent.com/2121359/163894636-294075f7-bb84-462a-bcdf-1acf1be9aa18.png)

![image](https://user-images.githubusercontent.com/2121359/163894993-7fc30465-1324-4256-9c21-09ea25d558f3.png)

Additionally, with the move to slash commands, it now allows users to autofill certain parameters (such as log channel and role selection) for commands which require it.

![image](https://user-images.githubusercontent.com/2121359/163894893-fb8b51a1-1128-4408-8a33-c033ad18b637.png)

The move to slash commands makes the bot much easier to use for both users wishing to verify their memberships, as well as moderators and administrators wanting to manage their server.

RisingFog committed:
Initial slash command implementation

RisingFog committed:
slash commands must all be lowercase

RisingFog committed:
fix missing character in perm check function

RisingFog committed:
copy global commands to guild commands

RisingFog committed:
fix missing string declaration for flag arguments

RisingFog committed:
set explicit variable types for function arguments

RisingFog committed:
various fixes

RisingFog committed:
add auto-complete for members and roles

RisingFog committed:
move verify command to slash command

RisingFog committed:
fix set/delete membership functions

RisingFog committed:
remove error helpers for viewmembers and delmember

RisingFog committed:
add date formatting hint for addmember

RisingFog committed:
add vtuber autocomplete for various functions

RisingFog committed:
remove error helper for settings commands

RisingFog committed:
remove hidden commands comment

RisingFog committed:
show correct vtuber membership delete message

RisingFog committed:
make view membership return embeds instead

RisingFog committed:
limit slash commands to guild only

RisingFog committed:
add command to clear guild command syncs

RisingFog committed:
fix interaction dm check

RisingFog committed:
properly show vtuber names for multi-server

RisingFog committed:
capitalize each vtuber in multi-talent settings

RisingFog committed:
Fix multi-talent role removal

RisingFog committed:
Migrate to default permissions for slash commands

RisingFog committed:
Fix unintended verify use for multi-talent servers

RisingFog committed:
Fix slash commands called in threads

RisingFog committed:
Fix default permissions for slash commands

nonJerry commented on settings.py:
```diff
@@ -1,6 +1,8 @@
 # External
+from xmlrpc.client import Boolean
```
I think import wasn't intentionally?

RisingFog replied:
Nope

nonJerry commented on utility.py:
```diff
@@ -1,4 +1,5 @@
 #External
+from random import paretovariate
```
Again probably not intentional import I guess XD

RisingFog replied:
I don’t even remember adding these

RisingFog committed:
Merge branch 'master' into feature-slash-commands

RisingFog committed:
Remove randomly added imports

RisingFog committed:
Merge branch 'feature-slash-commands' of https://github.com/RisingFog/VeraBot into feature-slash-commands

RisingFog committed:
Add vtuber autocomplete for verify command

RisingFog committed:
properly name the verify autocomplete function

RisingFog committed:
Fix verify not sending ephermeral replies

RisingFog committed:
defer view membership response

RisingFog committed:
Fix viewmembers not clearing message on new embed

RisingFog committed:
Reduce embed description to 4000 characters

RisingFog committed:
Reduce embed description size to 2048

RisingFog committed:
Remove reference to removed $verify command

RisingFog committed:
Split long viewmembers lists into separate replies

RisingFog committed:
Filter autocomplete results based on user input

nonJerry committed:
Fix failing verification for single servers

If a name was given it only checked the multi servers and if no name was given it always went against None

nonJerry committed:
Add newer date taking precedence

nonJerry committed:
Fix handling on missing date

nonJerry committed:
Switch to official 2.0.0 of discord.py

nonJerry committed:
Add missing text on wrong boolean flag
